### ISSUE TITLE
Expired Item handling process

### ISSUE THREAD

**Issue by f-9t9it** [frappe/erpnext#19029]
Hello,
In ERPNext we cannot do anything with the expired item, which is not the real use case.
In most cases expired Items are either returned to the supplier or discarded out of the business.

Current ERPNext method wont allow either of the case. 
- In Purchase Invoice/Receipt As Purchase return it will not allow to return Expired Batch number.
- In Stock Entry It will not allow to Make Material Issue out of the warehouse/business.

This will affect the inventory. Only way to do the above is to adjust the expiry date of that batch to current date to make it as non-expired product and move it out of the business using the above mentioned methods, which is not practical in use cases with high volume of expired items.

Kindly allow returns or adjustments of expired batch items.

**Comment by aliriocastro** [comment]
Hello,

Yes, ERPNext should create an option allowing to use expired batch on stock transactions.

**Comment by aliriocastro** [comment]
@kennethsequeira do you have any updates about this issue? Thanks in advance.

**Comment by kennethsequeira** [comment]
@aliriocastro stock entry is still allowed. So you can make a material issue and get it out of the inventory

**Comment by f-9t9it** [comment]
@kennethsequeira But what about Purchase Return ? If user wants to return the item and get debit note ?
My proposal is to make batch Disable after a given period or manually disabled. and allow returns for expired and continue with no sales allowed for expired ones

**Comment by tecnofoodmx** [comment]
Yes' I've got the same issue.
I believe the right way to do so is to create a warehouse for returning expired items, to avoid expired items from mixing with "good" ones. 

After that, each user might decide what to do with it according to the nature of its process (i.e.: material issue, repack, reprocess, etc...)

**Comment by f-9t9it** [comment]
@tecnofoodmx 
Yes that is what we do but when it comes to returning the expired back to the supplier as Purchase return it wont allow. So there should be a way to make purchase return for expired items.
@sunhoww what is the possible solution for this?

**Comment by sunhoww** [comment]
> @sunhoww what is the possible solution for this?

One might be to conditionally ignore validation while creating _Stock Ledger Entry_.

However, as mentioned earlier in the thread, a possible workaround would to use _Stock Entry_ to remove the expired batches. And maybe handle return from the _Purchase Invoice_.

**Comment by f-9t9it** [comment]
> > @sunhoww what is the possible solution for this?
> 
> One might be to conditionally ignore validation while creating _Stock Ledger Entry_.
> 
> However, as mentioned earlier in the thread, a possible workaround would to use _Stock Entry_ to remove the expired batches. And maybe handle return from the _Purchase Invoice_.

This is just a workaround. But not practical is high volume returns.

**Comment by techomniflo** [comment]
Hii ,
ERPNext should create an option allowing to use expired batch on stock transactions.
I just want to know that Is ERPNext implementing this feature and if yes then when?

**Comment by kennethsequeira** [comment]
@ankush can we close the thread as this was fixed in later versions:

https://github.com/frappe/erpnext/pull/32370

https://github.com/frappe/erpnext/pull/31863
